tests/test_direct_solver_device.cu

```diff
@@ -0,0 +1,117 @@
+/*************************************************************************
+ * Copyright (c) 2018 by the mfmg authors                                *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * This file is part of the mfmg libary. mfmg is distributed under a BSD *
+ * 3-clause license. For the licensing terms see the LICENSE file in the *
+ * top-level directory                                                   *
+ *                                                                       *
+ * SPDX-License-Identifier: BSD-3-Clause                                 *
+ *************************************************************************/
+
+#define BOOST_TEST_MODULE direct_solver_device
+
+#include "main.cc"
+
+#include <mfmg/dealii_operator_device.cuh>
+#include <mfmg/exceptions.hpp>
+#include <mfmg/utils.cuh>
+
+#include <deal.II/lac/cuda_vector.h>
+#include <deal.II/lac/read_write_vector.h>
+#include <deal.II/lac/sparse_direct.h>
+
+BOOST_AUTO_TEST_CASE(direct_solver)
+{
+  // Create the cusolver_dn_handle
+  cusolverDnHandle_t cusolver_dn_handle = nullptr;
+  cusolverStatus_t cusolver_error_code;
+  cusolver_error_code = cusolverDnCreate(&cusolver_dn_handle);
+  mfmg::ASSERT_CUSOLVER(cusolver_error_code);
+  // Create the cusolver_sp_handle
+  cusolverSpHandle_t cusolver_sp_handle = nullptr;
+  cusolver_error_code = cusolverSpCreate(&cusolver_sp_handle);
+  mfmg::ASSERT_CUSOLVER(cusolver_error_code);
+  // Create the cusparse_handle
+  cusparseHandle_t cusparse_handle = nullptr;
+  cusparseStatus_t cusparse_error_code;
+  cusparse_error_code = cusparseCreate(&cusparse_handle);
+  mfmg::ASSERT_CUSPARSE(cusparse_error_code);
+
+  // Create the matrix on the host.
+  dealii::SparsityPattern sparsity_pattern;
+  dealii::SparseMatrix<double> matrix;
+  unsigned int const size = 30;
+  std::vector<std::vector<unsigned int>> column_indices(size);
+  for (unsigned int i = 0; i < size; ++i)
+  {
+    unsigned int j_max = std::min(size, i + 2);
+    unsigned int j_min = (i == 0) ? 0 : i - 1;
+    for (unsigned int j = j_min; j < j_max; ++j)
+      column_indices[i].emplace_back(j);
+  }
+  sparsity_pattern.copy_from(size, size, column_indices.begin(),
+                             column_indices.end());
+  matrix.reinit(sparsity_pattern);
+  for (unsigned int i = 0; i < size; ++i)
+  {
+    unsigned int j_max = std::min(size - 1, i + 1);
+    unsigned int j_min = (i == 0) ? 0 : i - 1;
+    matrix.set(i, j_min, -1.);
+    matrix.set(i, j_max, -1.);
+    matrix.set(i, i, 4.);
+  }
+  // Solve on the host.
+  dealii::SparseDirectUMFPACK direct_solver;
```

Comment:
I think an even easier (and maybe better) test to start with a random `x`, to `b=Ax`. This way, you have a reference solution and rhs, without using `UMFPACK`.